test/resizing.jl

```diff
@@ -30,6 +30,10 @@
     img1 = colorview(RGB, fill(0.9, 3, 5, 5))
     img2 = colorview(RGB, fill(N0f8(0.9), 3, 5, 5))
     @test isapprox(channelview(restrict(img1)), channelview(restrict(img2)), rtol=0.01)
+    # Non-1 indices
+    @test parent(restrict(OffsetArray(A, (-2,1,0)), 1)) == restrict(A, 1)
```

Comment:
In fact, there was a type instability due to [this method](https://github.com/JuliaImages/ImageTransformations.jl/blob/24912f9ea09b5ea7ff9285961fd1f0c6d0cc52de/src/resizing.jl#L13-L18), since `A` can change its type. Thanks for the suggestion!